server/src/main/java/org/elasticsearch/health/components/controller/ClusterCoordination.java

```diff
@@ -60,8 +65,32 @@ public static HealthComponentResult createClusterCoordinationComponent(
             }
         );
 
-        // Only a single indicator currently so it determines the status
-        final HealthStatus status = instanceHasMaster.status();
-        return new HealthComponentResult(NAME, status, Collections.singletonMap(INSTANCE_HAS_MASTER_NAME, instanceHasMaster));
+        var noEligibleMasterNodes = noEligibleMasterNodes(clusterState);
+        var lackOfQuorum = lackOfQuorum(clusterState);
+        final HealthStatus status = HealthStatus.merge(
+            Stream.of(instanceHasMaster, noEligibleMasterNodes, lackOfQuorum).map(HealthIndicatorResult::status)
+        );
+        return new HealthComponentResult(
+            NAME,
+            status,
+            Map.of(INSTANCE_HAS_MASTER_NAME, instanceHasMaster, NO_ELIGIBLE_MASTERS, noEligibleMasterNodes, LACK_OF_QUORUM, lackOfQuorum)
+        );
+    }
+
+    static HealthIndicatorResult noEligibleMasterNodes(ClusterState clusterState) {
+        for (DiscoveryNode node : clusterState.nodes()) {
+            if (node.getRoles().contains(DiscoveryNodeRole.MASTER_ROLE)) {
+                return HealthIndicatorResult.of(NO_ELIGIBLE_MASTERS, NAME, HealthStatus.GREEN, "There are eligible master nodes");
```

Comment:
I guess we should be consistent here. The indicator I committed has a period as I feel like a summary could potentially include multiple sentences and we should use periods here. So I guess my preference would be each summary is a grammatically correct sentence with a period. Although, I am open to other thoughts.

Reply:
Same comment applies to the other summaries.